docs/docs/feature-reference/adding-and-updating-custom-databases.md

```diff
@@ -97,9 +96,45 @@ tc.update_database(
 )
 ```
 
+Note that it may take up to 1 hour for the custom database to be ready to use.
+
+## Preserving File Structure
+**Use a prefix** as your `database_path` if you have a custom database that requires preserving directory structure. For 
+both local and S3 paths, this will place all files in their implied subdirectories.
+
+If a list of paths is used instead, file structure will **not be preserved**. All specified files will be placed into 
+the same directory.
+
+## Specifying `database_primary_name`
+If you are uploading multiple database files and your tool takes in a certain file or prefix 
+**instead of the whole directory** as its command-line argument, use `database_primary_name` to specify the name of 
+that file or prefix.
+
+For example, if your database is structured like:
+```text
+arbitrary-directory/
+├── db.files.1
+├── db.files.2
+└── db.files.3
+```
+and you would use this command to run the tool from the command line:
+```commandline
+example_tool --database arbitrary-directory/db.files --input example_input.fastq --output example_output_dir/
+```
+then you would specify `database_primary_name=db.files`, the **base name** of the database argument, in your 
+Toolchest call.
+
+### Specifying the directory itself
+Use `database_primary_name=None` for `add_database` if there is no primary name (i.e., the directory itself is the 
+command-line database).
 
+### Behavior differences
+Note that the argument's behavior is **slightly different** for `add_database` and `update_database`:
+- This argument is **mandatory** for `add_database`.
+- This argument is **optional** for `update_database`. If omitted, it assumes the same `database_primary_name` as the previous
+version of the custom database.
+  - If `database_primary_name` needs to be changed to `None`, contact Toolchest to correct this. 
```

Comment:
This is too long to understand, it reads like a spec rather than docs.

Try rephrasing to be simpler with examples and the "basic" use-case upfront. Remove the bolding, and dive into more specific examples later in the doc after covering the simple example case.